common/metrics/scope.go

```diff
@@ -30,15 +31,26 @@ type metricsScope struct {
 	scope          tally.Scope
 	defs           map[int]metricDefinition
 	isDomainTagged bool
+	cachedScopes   map[string]Scope
 }
 
 func newMetricsScope(scope tally.Scope, defs map[int]metricDefinition, isDomain bool) Scope {
-	return &metricsScope{scope, defs, isDomain}
+	return &metricsScope{
+		scope:          scope,
+		defs:           defs,
+		isDomainTagged: isDomain,
+		cachedScopes:   make(map[string]Scope),
```

Comment:
Is it possible to share the cache between `metricsClient` and `metricsScope`?

I think people use `metricsClient.Scope(idx, tags)` and `metricsScope.Tag(tags)` (if any existing scope is available) interchangeably. If the underlying cache are separate, then one will reuse the cached scope while the other one will create a new scope and I think it's a little bit confusing.